### PR TITLE
create functions for

### PR DESCRIPTION
**Pull request by fickryiman**
- add todo
- remove todo
- update todo

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,67 +1,153 @@
...
+    tasks = tasks.filter((todo, ref) => ref !== +index);
+    tasks = tasks.map((task, index) => ({ ...task, index: index + 1 }));
+    // const indexing = tasks.map(todo => {
+    //   const temporaryStorage = {};
+    //   temporaryStorage.index = todo.index > index ? todo.index - 1 : todo.index;
+    //   temporaryStorage.description = todo.description;
+    //   temporaryStorage.completed = todo.completed;
+    //   return temporaryStorage;
+    // });
+
```
Comment (Qoosim) [inline src/index.js]: - [OPTIONAL] Always try to remove commented code. They make your codebase looks untidy.

Comment (Qoosim) [inline src/index.js]: - [x] Kindly remove unused code from your codebase, they make your codebase looks untidy.

File: src/index.js [hunk 2]
```diff
@@ -1,67 +1,153 @@
...
+      temporaryStorage.completed = task.completed;
+      return temporaryStorage;
     });
-  });
+    localStorage.setItem('TODOS', JSON.stringify(tasks));
+  },
+  deleteTodo: (index) => {
+    let tasks = TodoCRUD.readTodos();
+    tasks = tasks.filter((todo, ref) => ref !== +index);
+    tasks = tasks.map((task, index) => ({ ...task, index: index + 1 }));
```
Comment (Qoosim) [inline src/index.js:44]: - [x] Great job implementing deleteTodo function. There's little issue here. In your previous milestone, you were supposed to have created a function for `Clear all completed` which will delete all the completed todos once that button is clicked. Also, it is beleived that that would copy to this new branch from where you made your PR for this milestone if your previous branch has been merged to the main branch. There's no function to delete all the completed todos. I can see that you implemented it for each todo. Kindly implement the function that will delete all the completed tasks. The remaining milestones cover only the tests for this module and `Clear all completed` is one of the functions you will test for. You're expected to have created that from your pervious milestone. Kindly do that please.

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @fickryiman, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct  GitHub Flow . :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (fickryiman) [reply src/index.js]: removed commented code

Comment (fickryiman) [reply src/index.js]: removed unused code

File: src/index.js (hunk 2 above)
Comment (fickryiman) [reply src/index.js:44]: create function for Clear all completed

**Review by levy002 — APPROVED** [review]
## Hello @fickryiman 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Status: APPROVED :fire:  :fire: 

## Highlights
- Well-detailed README :heavy_check_mark: 
- No linter errors :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**